bfs: new port

ylluminarious opened the pull request:
#### Description

This adds a new port for the [`bfs`](https://github.com/tavianator/bfs) program, a breadth-first version of the Unix `find` command.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.14.2 18C54
Xcode 10.1 10B61

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

ylluminarious commented:
Oops, I accidentally meant to commit a change which changed `configure {}` to `use_configure no`. I just force-pushed it... hope it didn't cause any problems.

mf2k commented on textproc/bfs/Portfile:
```diff
@@ -0,0 +1,29 @@
+version             1.3.2
+categories          textproc
+platforms           darwin freebsd linux
+license             BSD-0-clause
```
This is not a normal license phrase in Macports. I believe you want "BSD" here.

mf2k commented on textproc/bfs/Portfile:
```diff
@@ -0,0 +1,29 @@
+description         A breadth-first version of the UNIX find command.
+homepage            https://github.com/tavianator/bfs
+master_sites        ${homepage}/archive/
+distname            ${version}
```
Use the GitHub portgroup to simplify the Portfile.

mf2k commented on textproc/bfs/Portfile:
```diff
@@ -0,0 +1,29 @@
+                    sha256   45939e6c1921c90e5f159407f877178cc9376dc66e808dbef80600d7624cbbc2 \
+                    size     88162
+
+worksrcdir          ${name}-${version}
```
This line can be deleted once you switch to the GitHub portgroup.

mf2k commented on textproc/bfs/Portfile:
```diff
@@ -0,0 +1,29 @@
+                    size     88162
+
+worksrcdir          ${name}-${version}
+use_configure       no
```
You are going to need to make sure that you are [using the right compiler](https://trac.macports.org/wiki/UsingTheRightCompiler).

ylluminarious replied:
@mf2k I don't think this program cares very much about which compiler it uses. It's supposed to be portable, so the default one chosen by Macports should suffice.

mf2k replied:
You need to at least make sure the universal variant works or not. Without the configure phase, manual changes, as mentioned in the wiki page, need to happen to make it work.

ylluminarious replied:
@mf2k Ok, I added some code for the universal variant so that it can build a universal binary if chosen by the user. [According to the docs:](https://guide.macports.org/chunked/reference.variants.html#reference.variants.user-selected) "The [universal] variant can be overridden if the default code does not work."

Selecting `+universal` does not work on Mojave; it fails with this error:

```
Error: bfs cannot be installed for the configured universal_archs 'x86_64 i386' because it only supports the arch(s) ''.
```

I'm inclined to believe this error occurred because Mojave does not seem to play nice with universal binaries anymore. I don't know whether this warrants a note or not.

However, selecting `+universal` **does** work on High Sierra. It installed successfully and the output of `file /opt/local/bin/bfs` is:

```
/opt/local/bin/bfs: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [i386:Mach-O executable i386]
/opt/local/bin/bfs (for architecture x86_64):	Mach-O 64-bit executable x86_64
/opt/local/bin/bfs (for architecture i386):	Mach-O executable i386
```

ylluminarious commented:
@mf2k I implemented the changes you requested, except for the one about [using the right compiler.](https://github.com/macports/macports-ports/pull/3421#discussion_r247579262) I don't think that's an issue in this case.

ylluminarious commented on textproc/bfs/Portfile:
```diff
@@ -0,0 +1,30 @@
+
+github.setup        tavianator bfs 1.3.2
+
+categories          textproc
```
@tavianator Yes, I also considered putting it under `sysutils`... I wasn't sure which category was more appropriate. Perhaps we could use some feedback from @mf2k.

mf2k replied:
Let's go with sysutils then.

ylluminarious replied:
@mf2k Alright, done.

ylluminarious commented:
@mf2k I went ahead and added some code for running the tests and I also added a `build` argument for setting `CC`. I basically just implemented @tavianator's suggestions from [this comment.](https://github.com/tavianator/homebrew-tap/issues/1#issuecomment-455608457) The aforementioned tests didn't all pass on my computer, but there were only 2 failures so I think it's negligible. Plus, tests aren't required to pass anyway, as I understand things.

So, all in all I think this is good to go, unless you have any other concerns.

mf2k commented:
Thanks!

tavianator commented:
> The aforementioned tests didn't all pass on my computer, but there were only 2 failures so I think it's negligible.

@ylluminarious Feel like opening a bfs issue with those failures?

ylluminarious commented:
@tavianator Strange, after I installed `bfs` via the default Macports `rsync` source (instead of the test source on my local filesystem), the tests actually pass now. So I guess it's a non-issue now.
